### DISCUSSION TITLE
Versioning across the PROTEUS ecosystem

### DISCUSSION THREAD

**Discussion by timlichtenberg** [FormingWorlds/PROTEUS#265] [General]
In the last PROTEUS meeting @stefsmeets suggested to homogenise version numbers across all codes to use semantic version numbers instead of CalVer. My biggest issue with this is that we now already use CalVer, so transitioning to CalVer would make the version history go to lower numbers again. Is there a way to resolve this?

**Comment by stefsmeets** [comment]
I mostly suggested it for aragog. Given the choice I would not advise for CalVer over SemVer, but if it is already in place it makes sense to keep using it. In my view SemVer is more intuitive to work with and more forgiving.
For example, if you bork a release and need to have a new release on the same day, you would end up with 2024.11.19.1 or something. Whereas with SemVer you just bump the patch version.

Switching from CalVer to SemVer sounds like a complete pain that is better to avoid, especially if the package is already on PyPi. In the end it does not matter too much, as long it is clear how it works, you stick to it, and have a regular release schedule.

**Comment by timlichtenberg** [comment]
Alright, then for now let's keep the version systems in the respective modules that are already in place. For new modules or ones that do not yet have a versioning system, we can use SemVer, or whatever the respective main author prefers.
